main.go

```diff
@@ -0,0 +1,363 @@
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.schedule.createIndex({name: 1}, {unique: true});
+	err = db.C("schedule").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.scheduleEvent.createIndex({name: 1}, {unique: true});
+	err = db.C("scheduleEvent").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	db = mgo.Database{
+		Session: session,
+		Name:    "coredata",
+	}
+
+	err = db.UpsertUser(&mgo.User{
+		Username: "core",
+		Password: "password",
+		Roles: []mgo.Role{
+			mgo.RoleReadWrite,
+		},
+	})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.event.createIndex({"device": 1}, {unique: false});
+	err = db.C("event").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.reading.createIndex({"device": 1}, {unique: false});
+	err = db.C("reading").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.valueDescriptor.createIndex({name: 1}, {unique: true});
+	err = db.C("valueDescriptor").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	db = mgo.Database{
+		Session: session,
+		Name:    "rules_engine_db",
+	}
+
+	err = db.UpsertUser(&mgo.User{
+		Username: "rules_engine_user",
+		Password: "password",
+		Roles: []mgo.Role{
+			mgo.RoleReadWrite,
+		},
+	})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	db = mgo.Database{
+		Session: session,
+		Name:    "notifications",
+	}
+
+	err = db.UpsertUser(&mgo.User{
+		Username: "notifications",
+		Password: "password",
+		Roles: []mgo.Role{
+			mgo.RoleReadWrite,
+		},
+	})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.notification.createIndex({slug: 1}, {unique: true});
+	err = db.C("notification").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	transmission := mgo.Collection{
+		Database: &db,
+		Name:     "transmission",
+		FullName: "db.transmission",
+	}
+	err = transmission.Create(&mgo.CollectionInfo{})
+	if err != nil {
+		LoggingClient.Error("Error during execution: " + err.Error())
+	}
+
+	//db.subscription.createIndex({slug: 1}, {unique: true});
+	err = db.C("subscription").EnsureIndex(mgo.Index{Key: []string{"name"}, Name: "name_1", Unique: true})
```

Comment:
Subscription index is on "slug" field in original script, not "name
https://github.com/edgexfoundry/docker-edgex-mongo/blob/b88dad2df2562f515b62e0cb486db88385442658/init_mongo.js#L108

Reply:
I'm not sure I can set the `slug` field with our current driver. https://godoc.org/github.com/globalsign/mgo#Index

It seems like their `Index` type doesn't have a way to set it. Am I missing something?